snuba/migrations/runner.py

```diff
@@ -81,30 +87,39 @@ def get_status(migration_key: MigrationKey) -> Status:
         return migrations
 
     def run_migration(
-        self, migration_key: MigrationKey, reverse: bool = False, force: bool = False
+        self, migration_key: MigrationKey, *, force: bool = False
     ) -> None:
         """
         Run a single migration given its migration key and marks the migration as complete.
 
-        TODO: Ensure that blocking migrations are run with force.
+        Blocking migrations must be run with force.
         """
         migration_id = migration_key.migration_id
 
-        assert all(
-            cluster.is_single_node() for cluster in CLUSTERS
-        ), "Cannot run migrations for multi node clusters"
+        context = Context(
+            migration_id, logger, partial(self._update_migration_status, migration_key),
+        )
+        migration = get_group_loader(migration_key.group).load_migration(migration_id)
+
+        if migration.blocking and not force:
+            raise MigrationError("Blocking migrations must be run with force")
+
+        migration.forwards(context)
+
+    def reverse_migration(
+        self, migration_key: MigrationKey, *, force: bool = False
+    ) -> None:
+        migration_id = migration_key.migration_id
 
         context = Context(
             migration_id, logger, partial(self._update_migration_status, migration_key),
         )
         migration = get_group_loader(migration_key.group).load_migration(migration_id)
 
-        if reverse:
-            if not force:
-                raise MigrationError("Reverse migration must be run with 'force'")
-            migration.backwards(context)
-        else:
-            migration.forwards(context)
+        if not force:
+            raise MigrationError("Reverse migration must be run with force")
```

Comment:
I am not sure you need a `force` parameter for this method. This would be useful on a CLI, but since this method works in a way only and it fails if force is not passed, why requiring it at all ?

Reply:
Removed it from here, will add it before this gets called from the CLI eventually.